core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java

```diff
@@ -57,7 +59,7 @@ public static MessageReader parseStructuredOrBinaryMessage(
             return binaryMessageFactory.apply(SpecVersion.parse(specVersionUnparsed));
         }
 
-        return unknownMessageFactory.get();
+        throw CloudEventRWException.newOther(new IllegalStateException("Invalid content type or spec version"));
```

Comment:
There is a CloudEventRWException kind for it right? If not, can we create it?

Reply:
I will fix the dco. As for that exception, this call is creating that exception

Reply:
I will work on the test

Reply:
I would love to have a specific `CloudEventRWException` for this, adding a kind here: https://github.com/cloudevents/sdk-java/blob/master/api/src/main/java/io/cloudevents/rw/CloudEventRWException.java#L28